### DISCUSSION TITLE
성능 측면에서 클로저는 문제가 되지 않는 지 설명해주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#91] [클로저]
- 성능 측면에서 클로저는 문제가 되지 않는 지 설명해주세요.

**Comment by pipisebastian** [comment]
> * 성능 측면에서 클로저는 문제가 되지 않는 지 설명해주세요.

- 클로저는 렉시컬 환경을 계속 기억해야 하기에, 클로저 사용이 많아질 수록 메모리 사용량을 증가시킵니다. 
- 그러나, 클로저의 메모리 정보는, 렉시컬 환경을 다 기억하는 것이 아닌, 필요한 식별자, 즉 참조중인 식별자만 기억합니다.

**Comment by publdaze** [comment]
> - 성능 측면에서 클로저는 문제가 되지 않는 지 설명해주세요.

- 이론적으로 클로저는 상위 렉시컬 환경을 기억해야 해서 메모리 점유에 대한 우려가 있긴하지만
- 자바스크립트 엔진에서 사용되는 식별자만 기억하도록 최적화하여, 필요한 것에 대한 메모리만 사용하기 때문에 크게 걱정할 필요는 없을 것
